### PR TITLE
properly handle and interpret X-Forwarded-For header

### PR DESCRIPTION
**Pull request by melange396**
configuration and handling to prevent potential ip address spoofing when behind certain kinds of proxies.

see loquacious comments inside the PR for more details.

### CODE REVIEW

File: src/server/_config.py [hunk 1]
```diff
@@ -26,14 +26,37 @@
...
+
+it is important that REVERSE_PROXY_DEPTH is set accurately for two reasons...
+
+setting it too high (or to -1) will respect more of the entries in the "X-Forwarded-For" header than are appropriate.
+ this can allow remote addresses to be "spoofed" when a client fakes this header, carrying security/identity
+ implications.  in dev and testing, it is not particularly dangerous for this variable to be set to -1 (special case
+ for an "infinite" depth, where any and all proxy hops will be trusted).
+
+setting it too low can hinder logging accuracy -- that can cause an intermediate proxy IP address to be used as the
+ "real" client IP address.
```
Comment (krivard) [inline src/server/_config.py:53]: ?

```suggestion
 "real" client IP address, which could cause requests to be rate-limited inappropriately.
```

File: src/server/_common.py [hunk 2]
```diff
@@ -31,14 +31,29 @@ def _get_db() -> Connection:
...
         if "X-Forwarded-For" in req.headers:
-            return req.headers["X-Forwarded-For"].split(",")[
-                0
-            ]  # take the first (or only) address from the comma-sep list
+            full_proxy_chain = req.headers["X-Forwarded-For"].split(",")
+            # eliminate any extra addresses at the front of this list, as they could be spoofed.
+            if REVERSE_PROXY_DEPTH > 0:
+                depth = REVERSE_PROXY_DEPTH
+            else:
+                # special case for -1/negative: setting `depth` to 0 will not strip any items from the chain
```
Comment (krivard) [inline src/server/_common.py:46]:
```suggestion
                # special case for -1/0/negative: setting `depth` to 0 will not strip any items from the chain
```

**Review by krivard — APPROVED** [review]
👍 
Minor comment fixes for clarity, only if accurate

File: src/server/_common.py (hunk 2 above)
Comment (melange396) [reply src/server/_common.py:46]: it cant have a value of zero here because of the `if REVERSE_PROXY_DEPTH:` above

Comment (krivard) [reply src/server/_common.py:46]: 😱 

...that's probably fine

**Review by krivard — APPROVED** [review]
❤️ loquacious comments

**Pull request merged**